hercules/tools/resource_downloading/wind_solar_resource_downloader.py

```diff
@@ -0,0 +1,1047 @@
+
+            data_dict["coordinates"] = data_dict["coordinates"].reset_index(drop=True)
+
+        # Save variables to feather format
+        for var_name in original_var_names:
+            output_file = os.path.join(
+                output_dir, f"{filename_prefix}_{var_name}_{time_suffix}.feather"
+            )
+            data_dict[var_name].reset_index().to_feather(output_file)
+            print(f"Saved {var_name} data to {output_file}")
+
+        # Save coordinates
+        coords_file = os.path.join(output_dir, f"{filename_prefix}_coords_{time_suffix}.feather")
+        data_dict["coordinates"].reset_index().to_feather(coords_file)
+        print(f"Saved coordinates to {coords_file}")
+
+    except Exception as e:
+        print(f"Error downloading Open-Meteo data: {e}")
+        return {}
+
+    total_time = (time.time() - t0) / 60
+    decimal_part = math.modf(total_time)[0] * 60
+    print(
+        "Open-Meteo download completed in "
+        f"{int(np.floor(total_time))}:{int(np.round(decimal_part, 0)):02d} minutes"
+    )
+
+    # Create plots if requested
+    if plot_data and data_dict and "coordinates" in data_dict:
+        coordinates_array = data_dict["coordinates"][["lat", "lon"]].values
+        if plot_type == "timeseries":
+            plot_timeseries(
+                data_dict, variables, coordinates_array, f"{filename_prefix} Open-Meteo Data"
+            )
+        elif plot_type == "map":
+            plot_spatial_map(
+                data_dict, variables, coordinates_array, f"{filename_prefix} Open-Meteo Data"
+            )
+
+    return data_dict
+
+
+def plot_timeseries(data_dict: dict, variables: List[str], coordinates: np.ndarray, title: str):
+    """
+    Create time-series plots for the downloaded data.
+
+    Parameters:
+    -----------
+    data_dict : dict
+        Dictionary containing DataFrames for each variable
+    variables : List[str]
+        List of variables to plot
+    coordinates : np.ndarray
+        Array of coordinates for the data points
+    title : str
+        Title for the plots
+    """
+
+    n_vars = len(variables)
+    if n_vars == 0:
+        return
+
+    # Create subplots based on number of variables
+    fig, axes = plt.subplots(n_vars, 1, figsize=(12, 4 * n_vars), sharex=True)
+    if n_vars == 1:
+        axes = [axes]
+
+    for i, var in enumerate(variables):
+        if var in data_dict:
+            df = data_dict[var]
+
+            # Plot all time series (one for each spatial point)
+            for col in df.columns:
+                axes[i].plot(df.index, df[col], alpha=0.7, linewidth=0.8)
+
+            axes[i].set_ylabel(get_variable_label(var))
+            axes[i].set_title(f"{var.replace('_', ' ').title()}")
+            axes[i].grid(True, alpha=0.3)
+
+    axes[-1].set_xlabel("Time")
+    plt.suptitle(f"{title} - Time Series", fontsize=14, fontweight="bold")
+    plt.tight_layout()
+    plt.show()
```

Comment:
Consider removing and having only in runscripts

Reply:
Removed here too